### PR TITLE
Run flake8 linter on python code

### PR DESCRIPTION
**Pull request by cclauss**
* Add Python 2.7 and 3.7 runs to Travis CI in alignment with travis-ci/travis-ci#9069
* Add [flake8](http://flake8.pycqa.org) tests to look for Python syntax errors and undefined names

__E901,E999,F821,F822,F823__ are the "showstopper" flake8 issues that can halt the runtime with a SyntaxError, NameError, etc.  Most other flake8 issues are merely "style violations" -- useful for readability but they do not effect runtime safety.
* F821: undefined name `name`
* F822: undefined name `name` in `__all__`
* F823: local variable `name` referenced before assignment
* E901: SyntaxError or IndentationError
* E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by cclauss** [comment]
#474 incorporates all of these changes and adds flake8 tests to Travis CI on both Python 2 and Python 3.

**Comment by dceejay** [comment]
Does this need updating now I merged your other PR ? #473 ?

**Comment by cclauss** [comment]
Rebased.

File: hardware/unicorn/unihat.py [hunk 1]
```diff
@@ -6,6 +6,11 @@
 import os
 import unicornhat as UH
 
+try:
+    raw_input          # Python 2
+except NameError:
+    raw_input = input  # Python 3
+
 if sys.version_info >= (3,0):
```
Comment (dceejay) [inline hardware/unicorn/unihat.py]: hmm - guess this can now be removed ?

Comment (cclauss) [reply hardware/unicorn/unihat.py]: https://pypi.org/project/unicornhat is Python 3 compatible but I think that https://github.com/node-red/node-red-nodes/blob/master/hardware/unicorn/scripts/checklib.js#L4 still stands in our way.

**Pull request merged**